### PR TITLE
Added support for eventlog

### PR DESCRIPTION
**Pull request by michalpristas**
Eventlog is append-only log. 
Will be used in
- Olympus as a log of backed modules
- Proxy as a log of backed modules
- Proxy as a log of registered cache misses (clearable after read)

Multiple backing storages should be supported, for now, mongo is implemented. 
Preparation for Olympus reader, which is remote reader of Olympus event log, is in place without implementation, this implementation is for proxies to read logs from Olympus, and Olympus instances between each other (Pull synchronization)

### CODE REVIEW

File: pkg/eventlog/multireader.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package eventlog
+
+import "github.com/gomods/athens/pkg/storage"
+
+type multiReader struct {
+	logs    []PointedLog
+	checker storage.Checker
+}
+
+// PointedLog is collection of event logs with specified starting pointers used by ReadFrom function.
```
[the remark this answers is not in the document]
Comment (michalpristas) [reply pkg/eventlog/multireader.go]: I like it better than what i came up with

File: pkg/eventlog/eventlog.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+	// ReadFrom reads all events from the log starting at event with specified id (excluded).
+	// If id is not found behaves like Read().
+	ReadFrom(id string) ([]Event, error)
+}
+
+// Appender is writer to append only event log.
+type Appender interface {
+	// Write appends Event to event log and returns its ID.
+	Write(event Event) (string, error)
```
Comment (arschles) [inline pkg/eventlog/eventlog.go]: nit: can we call this `Append` instead? that name seems to more accurately describe what it's doing

File: pkg/eventlog/mongo/mongo.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	c   string // collection
+	url string
+}
+
+// NewLog creates log reader from remote log
+func NewLog(url string) (*Log, error) {
+	return NewLogWithCollection(url, "eventlog")
+}
+
+// NewLogWithCollection creates log reader from remote log
```
Comment (arschles) [inline pkg/eventlog/mongo/mongo.go]: seems to return more than a reader - a writer and an appender, right?

File: pkg/eventlog/mongo/mongo.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		d:   "athens",
+	}
+	return m, m.Connect()
+}
+
+// Connect establishes a session to the mongo cluster.
+func (m *Log) Connect() error {
+	s, err := mgo.Dial(m.url)
+	if err != nil {
+		panic(err)
```
Comment (arschles) [inline pkg/eventlog/mongo/mongo.go]: can you return the error here? would rather do that than panic in a non-root stack frame

**Pull request merged**